### ISSUE TITLE
Multiple issues

### ISSUE THREAD

**Issue by natewarr** [GorillaStack/auto-tag#2]
I had a hard time getting this to work. I set up the appropriate roles but it keeps giving same error: 
```json

{
    "errorMessage": "Cannot read property 'bucket' of undefined",
    "errorType": "TypeError",
    "stackTrace": [
        "/var/task/aws_cloud_trail_listener.js:72:38",
        "Array.map (native)",
        "/var/task/aws_cloud_trail_listener.js:71:56",
        "new Promise (/var/task/node_modules/babel/node_modules/babel-core/node_modules/core-js/modules/es6.promise.js:178:7)",
        "AwsCloudTrailListener.retrieveLogFileDetails (/var/task/aws_cloud_trail_listener.js:69:14)",
        "callee$2$0$ (/var/task/aws_cloud_trail_listener.js:40:28)",
        "tryCatch (/var/task/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:61:40)",
        "GeneratorFunctionPrototype.invoke [as _invoke] (/var/task/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:323:22)",
        "GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/var/task/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:94:21)",
        "onFulfilled (/var/task/node_modules/co/index.js:65:19)"
    ]
}

```

I probably bit off more than I could chew, never developing in node.js before, but I couldn't run the main.js file using provided instructions. After installing the babel-preset-es2015 package, i kept getting errors trying to find babel/polyfill.

**Comment by MaheshBeri** [comment]
@natewarr Could you get this working?

i got similar error. Very hard to troubleshoot as no experience in nodeJS + Lambda.
I cannot get this function getting invoked automatically. Trying to test manually.

`{
  "errorMessage": "Cannot read property 'map' of undefined",
  "errorType": "TypeError",
  "stackTrace": [
    "/var/task/aws_cloud_trail_listener.js:71:55",
    "new Promise (/var/task/node_modules/babel/node_modules/babel-core/node_modules/core-js/modules/es6.promise.js:178:7)",
    "AwsCloudTrailListener.retrieveLogFileDetails (/var/task/aws_cloud_trail_listener.js:69:14)",`

**Comment by natewarr** [comment]
@MaheshBeri Nope. I gave up on this, I'm planning to try [capitalone/cloud-custodian](https://github.com/capitalone/cloud-custodian) to do this. It might require a pretty simple custom module, using cloudtrail logs to add a tag for the user that created the object.

**Comment by aae42** [comment]
it looks like i'm also having similar issues here

i get "Cannot read propery 'map' of undefined",

Is GorillaStack & @em0ney still supporting this? I guess i'll start looking at other options.

**Comment by em0ney** [comment]
Hey @aae42 @natewarr @MaheshBeri - still supported, just have been low on time recently.

Going to look into it this afternoon and do some improvements to the README, as well as provide a CloudFormation template for the setup of the lambda function and relevant roles.

**Comment by MaheshBeri** [comment]
@em0ney Thanks for update.Looking forward for information. 
I tired to run main.js but despite of lot of effort could not setup babel /node JS. 
Waiting for ways to get it working ..

**Comment by MaheshBeri** [comment]
@em0ney I saw that the Lambda function does not have in the policy for assume-role mentioned in documentation "Assuming the auto-tag role on targeted accounts". So wondering if it will work with incomplete policy. Or am I missing something?

**Comment by em0ney** [comment]
Hi @MaheshBeri 

I am almost finished a cloudformation template that will create the roles, bucket, cloudtrail, lambda and everything else required.  Will make setup much easier!

Also I have added a Gruntfile.js to make it easier to run tasks like transpilation (babel) etc on the project.

Looks like I will be pushing to github later this week.
Sorry for the time on this one, only have out of hours to spend on this item :)

Elliott

**Comment by MaheshBeri** [comment]
Hi @em0ney ,
While I am waiting for your next build, here is update.
I can now get function invoked (overcame issues with trust relationship etc). However Lambda function fails to get right object (which are coming as null). Not sure how to proceed now.
Thanks.

`2016-05-06T09:36:39.333Z 07c41523-136e-11e6-88d6-9136c8e2de5d TypeError: Cannot read property 'instancesSet' of null at AutotagEC2Worker.getInstanceId (/var/task/workers/autotag_ec2_worker.js:87:41) at callee$2$0$ (/var/task/workers/autotag_ec2_worker.js:54:51) at tryCatch (/var/task/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:61:40) at GeneratorFunctionPrototype.invoke [as _invoke] (/var/task/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:323:22) at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/var/task/node_modules/babel/node_modules/babel-core/node_modules/regenerator/runtime.js:94:21) at onFulfilled (/var/task/node_modules/co/index.js:65:19) at run (/var/task/node_modules/babel/node_modules/babel-core/node_modules/core-js/modules/es6.promise.js:89:39) at /var/task/node_modules/babel/node_modules/babel-core/node_modules/core-js/modules/es6.promise.js:100:28 at flush (/var/task/node_modules/babel/node_modules/babel-core/node_modules/core-js/modules/$.microtask.js:17:13) at nextTickCallbackWith0Args (node.js:415:9)`

**Comment by em0ney** [comment]
hey @natewarr @MaheshBeri @aae42,

Just published a new version 0.3.0 - check out the README.  Now we have a CloudFormation template to create all the resources required for AutoTag.

Hope this helps everyone get this set up correctly.  Apologies for any confusions with the old version of the README
